### ISSUE TITLE
BulkMerge Fails

### ISSUE THREAD

**Issue by sureshmohan251** [NorthernLight1/N.EntityFramework.Extensions#80]
I have a entity framework code first project . The entity has been defined with primary key and database generation option is set to none. If I try to bulk merge on this entity its failing with error **BulkMerge requires that the entity have a primary key or the Options.MergeOnCondition must be set.**

InventoryItem model
```
 public class InventoryItem
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int Id { get; set; }
        [StringLength(250)]
        public string Sku { get; set; }
        public int Quantity { get; set; }
  }
```

 Bulk merge method 
```
public void InventoryBulkMerge(List<InventoryItem> entities)
        {
            Context.BulkMerge(entities);
        }
```

**BulkMerge requires that the entity have a primary key or the Options.MergeOnCondition must be set.**
   at N.EntityFramework.Extensions.DbContextExtensions.InternalBulkMerge[T](DbContext context, IEnumerable`1 entities, BulkMergeOptions`1 options)
   at N.EntityFramework.Extensions.DbContextExtensions.BulkMerge[T](DbContext context, IEnumerable`1 entities, BulkMergeOptions`1 options)
   at N.EntityFramework.Extensions.DbContextExtensions.BulkMerge[T](DbContext context, IEnumerable`1 entities)

**Comment by NorthernLight1** [comment]
surehmohan251

I have validated this issue with a new test case and I will post a fix soon.

**Comment by NorthernLight1** [comment]
The issue has been resolved and I have added test cases for it as well. Install the latest version and let me know how it works for you.

**Comment by sureshmohan251** [comment]
Primary Key issue is resolved but getting another error for the column with has name attribute.  

```
[Column("FeedStatus")]
        [StringLength(25)]
        public string FeedStatusString
        {
            get { return FeedStatus.ToString(); }
            private set { FeedStatus = value.ParseEnum<FeedStatus>(); }
        }
```

**Invalid column name 'FeedStatusString'.**
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   at System.Data.Common.DbCommand.ExecuteReader()
   at N.EntityFramework.Extensions.DbContextExtensions.BulkQuery(DbContext context, String sqlText, SqlConnection dbConnection, SqlTransaction transaction, BulkOptions options)
   at N.EntityFramework.Extensions.DbContextExtensions.InternalBulkMerge[T](DbContext context, IEnumerable`1 entities, BulkMergeOptions`1 options)
   at N.EntityFramework.Extensions.DbContextExtensions.BulkMerge[T](DbContext context, IEnumerable`1 entities, BulkMergeOptions`1 options)
   at N.EntityFramework.Extensions.DbContextExtensions.BulkMerge[T](DbContext context, IEnumerable`1 entities)

**Comment by sureshmohan251** [comment]
Its fixed in 1.6.5. Thanks !
